src/id3_variants_training/ConfusionMatrix.py

```diff
@@ -15,81 +15,75 @@ def __init__(self, file_path, local):
         predicted is X axis
 
         Args:
-            file_path (str): Path to json file that contains the variant ranges
-            local (bool): Flag to determine whether or not to read locally or from a server
+            api (LOCAL_API | GA4GH_API): API object that is used to interact with the virtual API
 
         Attributes:
-            api (API): API object that is used to interact with the virtual API
+            api (LOCAL_API | GA4GH_API): API object that is used to interact with the virtual API
             root_node (Node): Creates the root node of the tree to be added upon
 
             length (int): length of all the ancestries
-            ancestry_list (list): list of all ancestries without repetitions
             conf_matrix (list): the confusion matrix based on the ID3 classifier
             diagonal_sum (int): sum of the diagonals within the matrix
             total (int): total sum of all values in matrix
-        '''
-        super(ConfusionMatrix, self).__init__(file_path, local)
-
-        # modify API for conf_matrix
-        self.api = LOCAL_API(file_path, conf_matrix=True)
-
-        # initialize ID3 algorithm
-        subset = self.api.get_target_set()
-        self.root_node = ID3_Node('root', subset, True)
-        self.ID3(self.root_node)
+        """
+        super(ConfusionMatrix, self).__init__(api)
 
         # create conf_matrix and calculate useful attributes
         self.length = len(self.api.ancestry_list)
-        self.conf_matrix = [ [0 for x in range(0, self.length)] for y in range(0, self.length) ]
+        self.conf_matrix = [[0] * self.length for _ in range(self.length)]
```

Comment:
You can also do `[ [0]*self.length ]*self.length ]`.  Even easier is numpy arrays but that's probably a big dependency to bring in here where we're not really doing a lot of math.

Reply:
That approach doesn't work because each row will have the same memory address. So if you modify index 0 of the first row all the other rows will also be modified.

https://stackoverflow.com/a/44382900

Reply:
Ah, right you are

Reply:
Ah - ok, I also see that we have numpy installed as a requirement.  Let's use it this way then, and in a future refactoring of Confusion matrix we'll use numpy arrays which will simplify the sums we have to do in other methods.